org.eclipse.jdt.ui/ui/org/eclipse/jdt/internal/ui/text/correction/QuickAssistProcessor.java

```diff
@@ -981,252 +979,19 @@ public static boolean getRemoveVarOrInferredLambdaParameterTypes(IInvocationCont
-		ReturnType returnType= new ReturnType();
-		if (variableType instanceof ParameterizedType) {
-			variableType= (Type) ((ParameterizedType) variableType).typeArguments().get(0);
-			ITypeBinding returnTypeBinding= variableType.resolveBinding();
-			if (returnTypeBinding != null) {
-				if (returnTypeBinding.isCapture()) {
-					returnType.binding= returnTypeBinding.getErasure();
-					returnType.type= importRewrite.addImport(returnTypeBinding.getErasure(), ast);
-				} else if (returnTypeBinding.isWildcardType()) {
-					returnType.binding= returnTypeBinding.getBound();
-					returnType.type= importRewrite.addImport(returnTypeBinding.getBound(), ast);
-				} else {
-					returnType.type= importRewrite.addImport(returnTypeBinding, ast);
-					returnType.binding= returnTypeBinding;
-				}
-			}
-		}
-		return returnType;
-	}
-
-	private static Block getNewReturnBlock(AST ast, ITypeBinding returnTypeBinding) {
-		Block newBlock= ast.newBlock();
-		if (!"void".equals(returnTypeBinding.getName())) { //$NON-NLS-1$
-			ReturnStatement newReturnStatement= ast.newReturnStatement();
-			String bName= returnTypeBinding.getBinaryName();
-			if ("Z".equals(bName)) { //$NON-NLS-1$
-				newReturnStatement.setExpression(ast.newBooleanLiteral(false));
-			} else if (returnTypeBinding.isPrimitive()) {
-				newReturnStatement.setExpression(ast.newNumberLiteral());
-			} else if ("java.lang.String".equals(bName)) { //$NON-NLS-1$
-				newReturnStatement.setExpression(ast.newStringLiteral());
-			} else {
-				newReturnStatement.setExpression(ast.newNullLiteral());
-			}
-			newBlock.statements().add(newReturnStatement);
-		}
-		return newBlock;
-	}
-
 	private static boolean getAddMethodDeclaration(IInvocationContext context, ASTNode covering, Collection<ICommandAccess> resultingCollections) {
-		CompilationUnit astRoot= context.getASTRoot();
-		ExpressionMethodReference methodReferenceNode= covering instanceof ExpressionMethodReference
-				? (ExpressionMethodReference) covering
-				: ASTNodes.getParent(covering, ExpressionMethodReference.class);
-		if (methodReferenceNode == null) {
-			return false;
-		}
-		boolean addStaticModifier= false;
-		TypeDeclaration typeDeclaration= ASTNodes.getParent(methodReferenceNode, TypeDeclaration.class);
-
-		if (QuickAssistProcessorUtil.isTypeReferenceToInstanceMethod(methodReferenceNode)) {
-			String methodReferenceQualifiedName= ((Name) methodReferenceNode.getExpression()).getFullyQualifiedName();
-			String typeDeclarationName= astRoot.getPackage().getName().getFullyQualifiedName() + '.' + typeDeclaration.getName().getFullyQualifiedName();
-			if (!methodReferenceQualifiedName.equals(typeDeclarationName)
-					&& !methodReferenceQualifiedName.equals(typeDeclaration.getName().getFullyQualifiedName())) {
-				// only propose for references in same class
-				return false;
-			}
-			addStaticModifier= true;
+		if (resultingCollections == null) {
```

Comment:
You should do some form of verification before returning true.  A number of quick assists actually do the work and check if the fix is null or not in concert with the the resultingCollections check for null.  If you don't want to create the fix, you should at least add a method that will do some form of pre-verification.  For example, the original code starts by setting the methodReferenceNode field and returning false if it is null.  There are other returns of false below that that could be done quickly.

Reply:
Agree, i think i found similar cases while looking at some previously changed quick fixes as well. But my bad I didn't consider here when i did this change. 

I think i could extract code in the CoreFix class and use that here and also in createMethod. Will do that changes tomorrow and push a fix.

Reply:
@gayanper Any progress?  I can merge once the mentioned code change is done.

Reply:
@jjohnstn i was busy last few days. I will try to fix this today or tomorrow.